### PR TITLE
Update Chip component styles with wp-admin colors

### LINKED ISSUE #6794: Chip component alignment
### Description
The great chip alignment has begun! Following the design proposed in p6riRB-8NZ-p2, the status chips in WooPayments will be unified in color, shape, and typography. Eventually, this update will be made across all of Woo admin, and a proposal will be made for inclusion in the Gutenberg library to achieve a holistic admin experience.

This update to the status chip in WooPayments is important to help users focus on critical information and prevent them from overlooking important actions to take on their disputes, transactions, and other tasks.

### Acceptance criteria
The updated `Chip` design should replace all instances of the `Chip` found in WooPayments. This includes but isn't limited to the following placements:
- Disputes list
- Payment details
- Deposits list
- Payments overview (Deposits card, Account card)
- [x] #6818

Note that this doesn't include any updates to the `Pill` component, which is usually a rounded pill shape with border outline used for marketing and informational purposes in announcements and settings.

### Designs

🎨 Status chip component in Figma: 4Lj84b3GdHAWVPFZmuhu8f-fi-1748_72561

Atoms

Color can be used to indicate success, failure, that user needs to take action, and/or simply convey the process or condition of an item. All colors pass AAA contrast requirements. The color system uses consistently wp-0 (background) and wp-70 (text) values for gray, yellow, red, green, and blue.

The text size, padding, and aspect ratio of the variants remain consistent with the `Chip` component in Woo Admin, however this version would extend the existing Chip component to include a variation with no close button.

@Jinksi Let me know if I can provide any more details! @souravdebnath1986, Eric mentioned that this update could be included as part of the Disputes scope of work as this will be a massive improvement to the experience.

cc @nagpai @kalessil @mattallan @dwainm as you've been part of the conversation in the past and your feedback is valued here!

### LINKED ISSUE #6938: "Paid" status chip in Transaction details should map to same green used in "Paid" status in Deposits
### Description
As mentioned in #6794 there's a discrepancy where the `Paid` status chip is mapped to different colors between the Deposits and Transactions areas.

Transactions Details
image

Deposits
image

Even though the meaning of "Paid" is different in each of these contexts, we feel the status label should reflect the same color to align the visual meaning of "Paid" for merchants.

### Solution

image

Since green offers a moment of celebration and success, we think the `Paid` status chip should be mapped to the green color throughout. So the gray color in the transaction details should be replaced with green.

@Jinksi I'll ping you on this one, but feel free to recommend others for input.

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #6794
Fixes #6938 

### Changes proposed in this Pull Request

* Standard WP-admin colours have been added to `client/stylesheets/abstracts/_colors.scss` for use in `Chip` component and elsewhere
	* See Slack discussion about wp-admin color definitions: p1689649184093919-slack-CGGCLBN58
	* See [Standardization of WP-Admin colors in WordPress 5.7 – Make WordPress Core](https://make.wordpress.org/core/2021/02/23/standardization-of-wp-admin-colors-in-wordpress-5-7/) for details about standard colours
* `Chip` component styles have been updated to match design in `client/components/chip/style.scss`
* `isCompact` variant of chip has been removed as it is no longer required, as per design
* TransactionStatusChip specific styles have been removed, as they are now redundant: `client/components/transaction-status-chip/index.tsx`
* Test snapshots have been updated to reflect the changes above

**`Chip` components in updated in this PR:**
* Dispute List
* Dispute Details
* Transaction Details
* Transaction Status Chips
* Account Overview
* Capital Loan Status Chips

**Not in this PR:** Deposits status chips are being updated to use the `Chip` component in PR #6820.

---

### Design

See status chip component in Figma: 4Lj84b3GdHAWVPFZmuhu8f-fi-1748_72561

254060715-ce2e8d6b-753a-43a1-9ac0-6f2f9e33e27c

#### This PR's implementation

image

### Before/After

**Dispute List**

<div style="display: flex; align-items: center">
dispute-details-before
dispute-details-after
</div>

---

**Dispute Details**

<div style="display: flex; align-items: center">

</div>

---

**Transaction Details / Payment Status Chips**

<div style="display: flex; align-items: center">

</div>

<div style="display: flex; align-items: center">

</div>

---

**Transaction Status Chips**

Note: these chips had custom styles applied which are removed in this PR to align with a single Chip design – `client/components/transaction-status-chip/index.tsx`.

<div style="display: flex; align-items: center">
image
image
</div>

---

**Account Overview**

<div style="display: flex; align-items: center">

</div>

---

**Capital Loan Status Chips**

<div style="display: flex; align-items: center">

image

image
</div>

---

**Deposits**

Deposits status chips are being updated to use the `Chip` component in PR #6820.

---

### Testing instructions

* Checkout this branch and run `npm run build:client`.
* View the updated chip designs on various screens shown above.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6796` or branch name `update/6794-chip-component` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/6794-chip-component) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3cebd7a64645a85805e94f60bb4fd4cf384ab22d
- Build time: 2023-08-14 00:27:23 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/chip/style.scss [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+		background: $wp-red-0;
+		color: $wp-red-70;
 	}
 
 	&.chip-warning {
-		background: $studio-yellow-5;
-		color: $studio-yellow-50;
-	}
-
-	&.is-compact {
```
Comment (Jinksi) [inline client/components/chip/style.scss:35]: `is-compact`/`isCompact` variant has been removed as it is no longer required as per designs.

File: client/components/transaction-status-chip/style.scss
Comment (Jinksi) [inline client/components/transaction-status-chip/style.scss:1]: Custom Transaction Status Chip style have been removed as they are no longer required.

These styles were implementing the updated design of the `Chip` component that are part of this PR.

File: client/stylesheets/abstracts/_colors.scss [hunk 2]
```diff
@@ -3,3 +3,66 @@
 @import 'node_modules/@automattic/color-studio/dist/color-variables.scss';
 @import 'node_modules/@wordpress/base-styles/colors.native.scss';
 @import 'node_modules/@wordpress/base-styles/colors.scss';
+
+// WP-admin colors: https://make.wordpress.org/core/2021/02/23/standardization-of-wp-admin-colors-in-wordpress-5-7/
```
Comment (Jinksi) [inline client/stylesheets/abstracts/_colors.scss:7]: Adding all standardised WP-admin colors to allow code throughout WooPayments to utilise.

**Comment by Jinksi** [comment]
@elizaan36 This PR is ready for your feedback when you have the chance.

I currently have this branch running on the [Helix demo site](https://woo-delectably-sparkly-ninja.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview) for convenience.

Note this PR updates the instances of the `Chip` component as listed in the PR description. The Deposit Status `Pill`s are being changed to these `Chip`s in PR #6820.

**Review by brucealdridge — COMMENTED** [review]
Have reviewed code and tested. Changes are a big improvement to clarity of code.

You mention this fixes #6938, however the `Paid` chip is not showing green as is talked about in that issue.

image

**Comment by Jinksi** [comment]
Thanks @brucealdridge 🤝 I have addressed the `paid` payment status chip colour in 1030dca. See below for updated payment status chip colours, including dispute statuses.

**Before/After**

<div style="display: flex; align-items: center">


</div>

**Comment by elizaan36** [comment]
@Jinksi Nailed it! Looks good to me and thanks for sharing the demo site.

**Comment by Jinksi** [comment]
Awesome, thanks for your feedback @elizaan36!

**Review by brucealdridge — APPROVED** [review]
LGTM.
Confirmed recent changes, `Paid` now shows as 🟢 chip.

**Comment by Jinksi** [comment]
Thanks @brucealdridge !

**Pull request merged**